base/libgit2/libgit2.jl

```diff
@@ -212,7 +270,16 @@ function branch(repo::GitRepo)
     end
 end
 
-""" git checkout [-b|-B] <branch> [<start-point>] [--track <remote>/<branch>] """
+"""
+    branch!(repo::GitRepo, branch_name::AbstractString, commit::AbstractString="", track::AbstractString="", force::Bool=false, set_head::Bool=true)
+
+Equivalent to `git checkout [-b|-B] <branch_name> [<commit>] [--track <track>]`.
+Checkout a new git branch in the `repo` repository. `commit` is the [`GitHash`](@ref),
+in string form, which will be the start of the new branch. `track` is the name of the
+remote branch this new branch should track, if any. If empty (the default), no remote branch
+will be tracked. If `force` is `true`, branch creation will be forced. If `set_head` is
+`true`, as the branch creation finishes the branch head will be set as the HEAD of `repo`.
```

Comment:
after the branch creation finishes? doesn't make much sense for it to happen at exactly the same time